### PR TITLE
lib: Fix ListingTable nesting for non-expandable rows

### PR DESCRIPTION
**Pull request by martinpitt**
`<Tr>` always needs to be wrapped into a `<Tbody>`.

Clean up the unnecessary special cases and handle them uniformly.

Adjust the tests to fix the selectors, they previously relied on the
buggy behaviour.

----

Split out from PR #19509, this is already a very intrusive fix which should land separately.

### CODE REVIEW

**Comment by martinpitt** [comment]
The [other pixel changes](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19520-20231024-071602-0653d9d2-fedora-38-other/log.html) are correct -- these are now tbody's which look like separate "cards", as opposed to the adjoined "expandable row" ones.

Same for the [storage ones](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19520-20231024-071602-0653d9d2-fedora-38-storage/log.html), so I'll accept them.

File: pkg/lib/cockpit-components-inline-notification.jsx [hunk 1]
```diff
@@ -90,7 +90,7 @@ InlineNotification.propTypes = {
 export const ModalError = ({ dialogError, dialogErrorDetail, id, isExpandable }) => {
     return (
         <Alert id={id} variant='danger' isInline title={dialogError} isExpandable={!!isExpandable}>
-            { dialogErrorDetail && <p> {dialogErrorDetail} </p> }
+            { typeof dialogErrorDetail === 'string' ? <p>{dialogErrorDetail}</p> : dialogErrorDetail }
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-inline-notification.jsx:93]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19520-20231024-071602-0653d9d2-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
Argh, what a yak shave -- [NFS kdump is flaky on RHEL 8.9](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19520-20231024-085000-668e64da-rhel-8-9-expensive/log.html#4-1) and [RHEL 9.3](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19520-20231024-085000-668e64da-rhel-9-3-expensive/log.html). It's not new, it's a flake, but 3x affected retries expose it. On my todo list..

File: pkg/lib/cockpit-components-inline-notification.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-inline-notification.jsx:93]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19520-20231024-085000-668e64da-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
I can locally reproduce the NFS kdump failure rather well (about 1 in 2 runs). When that happens, the NFS server indeed does not have /srv/kdump/dumps/10.111.113.1* but it does have /srv/kdump/var/crash/10.111.113.1-2023-10-24-07:29:14/vmcore -- but I think that's still from the previous test which covers the default path.

The config is correct:
```
nfs 10.111.113.2:/srv/kdump
path dumps
```

and the dialog also confirms that it got the right path.

We recently had https://github.com/cockpit-project/bots/issues/5263 which came and go, and bz wasn't touched by anyone except us. That seems related..

File: test/common/storagelib.py [hunk 2]
```diff
@@ -121,18 +121,15 @@ def content_row_action(self, index, title, isExpandable=True):
         if isExpandable:
```
Comment (mvollmer) [inline test/common/storagelib.py:121]: I was hoping that we can get rid of the `isExpandable` distinction now.

File: test/common/storagelib.py [hunk 3]
```diff
@@ -121,18 +121,15 @@ def content_row_action(self, index, title, isExpandable=True):
...
-            btn = "#detail-content > .pf-v5-c-card > div > table > :nth-child(%d)" % index + f" td button:contains({title})"
+            btn = self.content_row_tbody(index) + f" td button:contains({title})"
         self.browser.click(btn)
 
     # The row might come and go a couple of times until it has the
     # expected content.  However, wait_in_text can not deal with a
     # temporarily disappearing element, so we use self.retry.
 
     def content_row_wait_in_col(self, row_index, col_index, val, isExpandable=True, alternate_val=None):
-        if isExpandable:
```
[the remark this answers is not in the document]
Comment (mvollmer) [reply test/common/storagelib.py:132]: Ah, let's not. The redesign is touching all of this code anyway.

[the remark this answers is not in the document]
Comment (martinpitt) [reply test/common/storagelib.py:132]: Right, that ought to work, the remaining difference should not be significant any more. Let's see how far it gets!

[the remark this answers is not in the document]
Comment (martinpitt) [reply test/common/storagelib.py:132]: ... and indeed that was unclean, as at least one of the functions didn't use the argument at all any more, and for the others the difference was just insignificant noise.

**Review by mvollmer — APPROVED** [review]

File: pkg/lib/cockpit-components-inline-notification.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-inline-notification.jsx:93]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19520-20231024-141450-05b107a4-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
I restored a bit of the selector in content_row_action() that was important for disambiguation, that should fix the tests. I also added an extra commit to make the KdumpNFS flakes easier to debug. @mvollmer PTAL?

File: pkg/lib/cockpit-components-inline-notification.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-inline-notification.jsx:93]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19520-20231025-041647-58976af1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
That [stratis rawhide failure](https://artifacts.dev.testing-farm.io/1b99754f-a96e-4073-a11f-b07202a82b86/) is entirely new and unrelated, I'll send a separate PR to fix it.

**Review by mvollmer — APPROVED** [review]

**Pull request merged**